src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -622,7 +643,10 @@ struct DecimalBinaryOperation
             if constexpr (op_case == OpCase::RightConstant)
             {
                 if ((*right_nullmap)[0])
+                {
+                    c[0] = ResultType();
```

Comment:
Seems like this is the problem. We should do it in a loop:
```suggestion
                    for (size_t i = 0; i < size; ++i)
                        c[i] = ResultType();
```